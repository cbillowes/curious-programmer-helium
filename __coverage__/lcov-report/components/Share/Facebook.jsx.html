<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for components/Share/Facebook.jsx</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../prettify.css" />
    <link rel="stylesheet" href="../../base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="../../index.html">All files</a> / <a href="index.html">components/Share</a> Facebook.jsx
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">0% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>0/9</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">0% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>0/2</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">0% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>0/4</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">0% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>0/8</span>
      </div>
    </div>
  </div>
  <div class='status-line low'></div>
<pre><table class="coverage">
<tr><td class="line-count quiet">1
2
3
4
5
6
7
8
9
10
11
12
13
14
15
16
17
18
19
20
21
22
23
24
25
26
27
28</td><td class="line-coverage quiet"><span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span></td><td class="text"><pre class="prettyprint lang-js"><span class="cstat-no" title="statement not covered" >import React, { Component } from "react"</span>
<span class="cstat-no" title="statement not covered" >import {</span>
  FacebookShareButton,
  FacebookShareCount,
  FacebookIcon
} from "react-share"
&nbsp;
class Facebook extends Component {
<span class="fstat-no" title="function not covered" >  render</span>() {
    const { url, title, description, size } = <span class="cstat-no" title="statement not covered" >this.props</span>
    const filter = <span class="cstat-no" title="statement not covered" ><span class="fstat-no" title="function not covered" >count</span> =&gt; (<span class="cstat-no" title="statement not covered" >count &gt; 0 ? count : "")</span></span>
    const renderShareCount = <span class="cstat-no" title="statement not covered" ><span class="fstat-no" title="function not covered" >count</span> =&gt; (</span>
<span class="cstat-no" title="statement not covered" >      &lt;div className="share-count"&gt;{filter(count)}&lt;/div&gt;</span>
    )
&nbsp;
<span class="cstat-no" title="statement not covered" >    return (</span>
      &lt;FacebookShareButton url={url} quote={`${title} | ${description}`}&gt;
        &lt;FacebookIcon round size={parseInt(size)} /&gt;
        &lt;FacebookShareCount url={url}&gt;
          {<span class="fstat-no" title="function not covered" >count</span> =&gt; <span class="cstat-no" title="statement not covered" >renderShareCount(count)}</span>
        &lt;/FacebookShareCount&gt;
      &lt;/FacebookShareButton&gt;
    )
  }
}
&nbsp;
export default Facebook
&nbsp;</pre></td></tr>
</table></pre>
<div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Sun Jan 06 2019 09:15:22 GMT+0200 (South Africa Standard Time)
</div>
</div>
<script src="../../prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="../../sorter.js"></script>
</body>
</html>
