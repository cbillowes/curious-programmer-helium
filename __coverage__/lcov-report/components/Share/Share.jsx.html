<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for components/Share/Share.jsx</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../prettify.css" />
    <link rel="stylesheet" href="../../base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="../../index.html">All files</a> / <a href="index.html">components/Share</a> Share.jsx
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">0% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>0/14</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>0/0</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">0% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>0/1</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">0% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>0/14</span>
      </div>
    </div>
  </div>
  <div class='status-line low'></div>
<pre><table class="coverage">
<tr><td class="line-count quiet">1
2
3
4
5
6
7
8
9
10
11
12
13
14
15
16
17
18
19
20
21
22
23
24
25
26
27
28
29
30
31
32
33</td><td class="line-coverage quiet"><span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span></td><td class="text"><pre class="prettyprint lang-js"><span class="cstat-no" title="statement not covered" >import React, { Component } from "react"</span>
<span class="cstat-no" title="statement not covered" >import Facebook from "./Facebook"</span>
<span class="cstat-no" title="statement not covered" >import Twitter from "./Twitter"</span>
<span class="cstat-no" title="statement not covered" >import Reddit from "./Reddit"</span>
<span class="cstat-no" title="statement not covered" >import GooglePlus from "./GooglePlus"</span>
<span class="cstat-no" title="statement not covered" >import LinkedIn from "./LinkedIn"</span>
<span class="cstat-no" title="statement not covered" >import Telegram from "./Telegram"</span>
<span class="cstat-no" title="statement not covered" >import WhatsApp from "./WhatsApp"</span>
<span class="cstat-no" title="statement not covered" >import config from "../../../data/SiteConfig"</span>
<span class="cstat-no" title="statement not covered" >import "./Share.scss"</span>
&nbsp;
class SocialLinks extends Component {
<span class="fstat-no" title="function not covered" >  render</span>() {
    const { slug, title, description } = <span class="cstat-no" title="statement not covered" >this.props</span>
    const url = `<span class="cstat-no" title="statement not covered" >${config.siteUrl}${config.pathPrefix}${slug}`</span>
    const size = <span class="cstat-no" title="statement not covered" >70</span>
&nbsp;
<span class="cstat-no" title="statement not covered" >    return (</span>
      &lt;div className="social-media"&gt;
        &lt;Facebook url={url} title={title} description={description} size={size} /&gt;
        &lt;Twitter url={url} title={title} size={size} /&gt;
        &lt;Reddit url={url} title={title} size={size} /&gt;
        &lt;GooglePlus url={url} title={title} size={size} /&gt;
        &lt;LinkedIn url={url} title={title} description={description} size={size} /&gt;
        &lt;Telegram url={url} size={size} /&gt;
        &lt;WhatsApp url={url} title={title} size={size} /&gt;
      &lt;/div&gt;
    )
  }
}
&nbsp;
export default SocialLinks
&nbsp;</pre></td></tr>
</table></pre>
<div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Sun Jan 06 2019 09:15:22 GMT+0200 (South Africa Standard Time)
</div>
</div>
<script src="../../prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="../../sorter.js"></script>
</body>
</html>
