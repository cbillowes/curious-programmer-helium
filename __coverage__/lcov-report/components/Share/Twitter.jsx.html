<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for components/Share/Twitter.jsx</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../prettify.css" />
    <link rel="stylesheet" href="../../base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="../../index.html">All files</a> / <a href="index.html">components/Share</a> Twitter.jsx
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">0% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>0/4</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>0/0</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">0% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>0/1</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">0% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>0/4</span>
      </div>
    </div>
  </div>
  <div class='status-line low'></div>
<pre><table class="coverage">
<tr><td class="line-count quiet">1
2
3
4
5
6
7
8
9
10
11
12
13
14
15
16
17
18
19
20
21
22</td><td class="line-coverage quiet"><span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span></td><td class="text"><pre class="prettyprint lang-js"><span class="cstat-no" title="statement not covered" >import React, { Component } from "react"</span>
<span class="cstat-no" title="statement not covered" >import {</span>
  TwitterShareButton,
  TwitterIcon
} from "react-share"
&nbsp;
class Twitter extends Component {
<span class="fstat-no" title="function not covered" >  render</span>() {
    const { url, title, size } = <span class="cstat-no" title="statement not covered" >this.props</span>
&nbsp;
<span class="cstat-no" title="statement not covered" >    return (</span>
      &lt;div className="social-links"&gt;
        &lt;TwitterShareButton url={url} title={title}&gt;
          &lt;TwitterIcon round size={parseInt(size)} /&gt;
        &lt;/TwitterShareButton&gt;
      &lt;/div&gt;
    )
  }
}
&nbsp;
export default Twitter
&nbsp;</pre></td></tr>
</table></pre>
<div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Sun Jan 06 2019 09:15:22 GMT+0200 (South Africa Standard Time)
</div>
</div>
<script src="../../prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="../../sorter.js"></script>
</body>
</html>
