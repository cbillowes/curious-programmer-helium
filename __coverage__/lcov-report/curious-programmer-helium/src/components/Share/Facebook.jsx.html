<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for curious-programmer-helium/src/components/Share/Facebook.jsx</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../../../prettify.css" />
    <link rel="stylesheet" href="../../../../base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../../../sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="../../../../index.html">All files</a> / <a href="index.html">curious-programmer-helium/src/components/Share</a> Facebook.jsx
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>7/7</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">50% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>1/2</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>4/4</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>6/6</span>
      </div>
    </div>
  </div>
  <div class='status-line high'></div>
<pre><table class="coverage">
<tr><td class="line-count quiet">1
2
3
4
5
6
7
8
9
10
11
12
13
14
15
16
17
18
19
20
21
22
23
24
25
26
27
28</td><td class="line-coverage quiet"><span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">4x</span>
<span class="cline-any cline-yes">4x</span>
<span class="cline-any cline-yes">4x</span>
<span class="cline-any cline-yes">4x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">4x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">4x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span></td><td class="text"><pre class="prettyprint lang-js">import React, { Component } from "react"
import {
  FacebookShareButton,
  FacebookShareCount,
  FacebookIcon
} from "react-share"
&nbsp;
class Facebook extends Component {
  render() {
    const { url, title, description, size } = this.props
    const filter = count =&gt; (count &gt; 0 ? <span class="branch-0 cbranch-no" title="branch not covered" >count </span>: "")
    const renderShareCount = count =&gt; (
      &lt;div className="share-count"&gt;{filter(count)}&lt;/div&gt;
    )
&nbsp;
    return (
      &lt;FacebookShareButton url={url} quote={`${title} | ${description}`}&gt;
        &lt;FacebookIcon round size={parseInt(size)} /&gt;
        &lt;FacebookShareCount url={url}&gt;
          {count =&gt; renderShareCount(count)}
        &lt;/FacebookShareCount&gt;
      &lt;/FacebookShareButton&gt;
    )
  }
}
&nbsp;
export default Facebook
&nbsp;</pre></td></tr>
</table></pre>
<div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Sat Jan 05 2019 22:01:48 GMT+0200 (South Africa Standard Time)
</div>
</div>
<script src="../../../../prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="../../../../sorter.js"></script>
</body>
</html>
