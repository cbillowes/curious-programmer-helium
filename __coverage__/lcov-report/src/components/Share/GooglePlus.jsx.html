<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for src/components/Share/GooglePlus.jsx</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../../prettify.css" />
    <link rel="stylesheet" href="../../../base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../../sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="../../../index.html">All files</a> / <a href="index.html">src/components/Share</a> GooglePlus.jsx
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>7/7</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">50% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>1/2</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>4/4</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>6/6</span>
      </div>
    </div>
  </div>
  <div class='status-line high'></div>
<pre><table class="coverage">
<tr><td class="line-count quiet">1
2
3
4
5
6
7
8
9
10
11
12
13
14
15
16
17
18
19
20
21
22
23
24
25
26
27
28</td><td class="line-coverage quiet"><span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">14x</span>
<span class="cline-any cline-yes">14x</span>
<span class="cline-any cline-yes">14x</span>
<span class="cline-any cline-yes">14x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">14x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">14x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span></td><td class="text"><pre class="prettyprint lang-js">import React, { Component } from "react"
import {
  GooglePlusShareButton,
  GooglePlusShareCount,
  GooglePlusIcon
} from "react-share"
&nbsp;
class GooglePlus extends Component {
  render() {
    const { url, size } = this.props
    const filter = count =&gt; (count &gt; 0 ? <span class="branch-0 cbranch-no" title="branch not covered" >count </span>: "")
    const renderShareCount = count =&gt; (
      &lt;div className="share-count"&gt;{filter(count)}&lt;/div&gt;
    )
&nbsp;
    return (
      &lt;GooglePlusShareButton url={url}&gt;
        &lt;GooglePlusIcon round size={parseInt(size)} /&gt;
        &lt;GooglePlusShareCount url={url}&gt;
          {count =&gt; renderShareCount(count)}
        &lt;/GooglePlusShareCount&gt;
      &lt;/GooglePlusShareButton&gt;
    )
  }
}
&nbsp;
export default GooglePlus
&nbsp;</pre></td></tr>
</table></pre>
<div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Sun Jan 06 2019 09:47:38 GMT+0200 (South Africa Standard Time)
</div>
</div>
<script src="../../../prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="../../../sorter.js"></script>
</body>
</html>
