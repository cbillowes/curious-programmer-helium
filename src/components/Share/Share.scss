@import "../../layout/layout.scss";

.social-media {
    display: flex;
    flex-direction: row;
    flex-wrap: wrap;
    justify-content: center;
    align-content: center;
    align-items: center;
    margin: 15px 0;

    > div {
        margin: 0.5em;
    }

    .social-icon {
        cursor: pointer;
    }

    .share-count {
        text-align: center;
    }

    svg:hover circle {
        fill: $background-lighter;
    }
}