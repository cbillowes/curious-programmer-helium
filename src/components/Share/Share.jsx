import React, { Component } from "react"
import Facebook from "./Facebook"
import Twitter from "./Twitter"
import Reddit from "./Reddit"
import GooglePlus from "./GooglePlus"
import LinkedIn from "./LinkedIn"
import Telegram from "./Telegram"
import WhatsApp from "./WhatsApp"
import config from "../../../data/SiteConfig"
import "./Share.scss"

class SocialLinks extends Component {
  render() {
    const { slug, title, description } = this.props
    const url = `${config.siteUrl}${config.pathPrefix}${slug}`
    const size = 50

    return (
      <div className="social-media">
        <Facebook url={url} title={title} description={description} size={size} />
        <Twitter url={url} title={title} size={size} />
        <Reddit url={url} title={title} size={size} />
        <GooglePlus url={url} title={title} size={size} />
        <LinkedIn url={url} title={title} description={description} size={size} />
        <Telegram url={url} size={size} />
        <WhatsApp url={url} title={title} size={size} />
      </div>
    )
  }
}

export default SocialLinks
