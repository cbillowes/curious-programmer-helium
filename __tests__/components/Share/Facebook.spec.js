import React from "react"
import renderer from "react-test-renderer"
import { mount } from "enzyme"
import Share from "../../../src/components/Share/Facebook"

describe("a facebook share component", () => {

  it("should render", () => {
    const element = <Share url="https://bit.ly/blog/this-is-a-post" description="Just a little write up." size="52" />
    const tree = renderer.create(element).toJSON()
    expect(tree).toMatchSnapshot()
  })

  it("should render its contents" , () => {
    const element = <Share url="https://bit.ly/blog/this-is-a-post" description="Just a little write up." size="52" />
    const component = mount(element)

    expect(component.html()).toContain(`SocialMediaShareButton--facebook`)
    expect(component.html()).toContain(`width="52" height="52"`)
    expect(component.html()).toContain(`social-icon--facebook`)
  })
})