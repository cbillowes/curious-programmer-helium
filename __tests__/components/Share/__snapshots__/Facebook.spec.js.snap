// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`a facebook share component should render 1`] = `
<div
  className="SocialMediaShareButton SocialMediaShareButton--facebook"
  onClick={[Function]}
  onKeyPress={[Function]}
  role="button"
  style={Object {}}
  tabIndex="0"
>
  <div
    style={
      Object {
        "height": 52,
        "width": 52,
      }
    }
  >
    <svg
      className="social-icon social-icon--facebook "
      height={52}
      viewBox="0 0 64 64"
      width={52}
    >
      <g>
        <circle
          cx="32"
          cy="32"
          fill="#3b5998"
          r="31"
          style={Object {}}
        />
      </g>
      <g>
        <path
          d="M34.1,47V33.3h4.6l0.7-5.3h-5.3v-3.4c0-1.5,0.4-2.6,2.6-2.6l2.8,0v-4.8c-0.5-0.1-2.2-0.2-4.1-0.2 c-4.1,0-6.9,2.5-6.9,7V28H24v5.3h4.6V47H34.1z"
          fill="white"
        />
      </g>
    </svg>
  </div>
  <div
    className="SocialMediaShareCount"
  />
</div>
`;
